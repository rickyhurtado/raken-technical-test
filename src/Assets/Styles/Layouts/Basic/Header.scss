$black: #8c8c8c;
$orange: #f26327;
$light-grey: #eaeaea;

@mixin menu-border-top {
  border-top: 1px solid #e7e7e7;
  &:first-child {
    border-top: none;
  }
}
@mixin dropdown-menu-top-arrow($borderColor, $borderWidth, $bottom, $right) {
  border: solid transparent;
  border-color: rgba(255, 255, 255, 0);
  border-bottom-color: #{$borderColor};
  border-width: #{$borderWidth};
  bottom: #{$bottom};
  content: " ";
  height: 0;
  position: absolute;
  right: #{$right};
  width: 0;
}

body.has-hover {
  .navbar {
    .navbar-brand, a {
      &:hover, &:active:hover, &:focus {
        color: $orange;
        text-decoration: none;
      }
    }
  }
}
.layout {
  > header.header {
    .navbar {
      background: #f3f3f3;
      border-color: $light-grey;
      .container {
        width: 100% !important;
        @media (max-width: 767px) {
          .navbar-header {
            background: #f3f3f3;
            border-bottom: 1px solid #e7e7e7;
            position: fixed;
            width: 100%;
            z-index: 99;
          }
        }
      }
      a {
        background: transparent !important;
        box-shadow: none;
        font-size: 12px;
        padding: 30px 25px;
        text-transform: uppercase;
        transition: all 0.3s;
      }
      .active {
        a {
          color: $orange;
        }
      }
      .navbar-brand {
        height: auto;
        font-size: 14px;
        font-weight: bold;
        &:focus {
          color: $black;
          &:hover {
            text-decoration: none;
          }
        }
      }
      .dropdown-menu {
        li {
          border-top: 1px solid #efefef;
          &:first-child {
            border-top: none;
          }
          &.active {
            a {
              color: $orange;
            }
          }
          a {
            line-height: 1;
            padding: 15px 40px 15px 0;
            position: relative;
            text-transform: uppercase;
            .notification-badge {
              background: $orange;
              border-radius: 50px;
              color: white;
              display: block;
              font-size: 8px;
              height: 20px;
              line-height: 2.7;
              position: absolute;
              right: 0;
              text-align: center;
              top: 10px;
              width: 20px;
            }
          }
          .sign-out-btn {
            background: transparent;
            border: none;
            border-radius: 0;
            display: block;
            font-size: 12px;
            text-align: left;
            text-transform: uppercase;
            transition: all 0.3s;
            padding: 15px 0;
            width: 100%;
            &:focus {
              outline: none;
            }
            &:hover {
              color: $orange;
            }
            @media (min-width: 768px) {
              padding-top: 15px;
              padding-bottom: 15px;
            }
          }
        }
      }
      @media (min-width: 768px) {
        .container {
          padding-right: 0;
        }
        .account-menu {
          border-left: 1px solid $light-grey;
          margin-left: 20px;
          .dropdown-toggle {
            background: url("/account-icon.png") no-repeat center !important;
            height: 0;
            overflow: hidden;
            padding: 80px 20px 0;
            &:hover {
              opacity: 0.7;
            }
          }
          .dropdown-menu {
            border: 1px solid #f3f3f3;
            border-radius: 10px;
            box-shadow: 0 4px 20px rgba(0, 0, 0, 0.175);
            padding: 0 20px;
            margin-top: 10px;
            li {
              &:first-child {
                border-top-left-radius: 10px;
                border-top-right-radius: 10px;
              }
              &.active {
                background: #d8d8d8;
                box-shadow: inset 0 5px 10px rgba(0, 0, 0, 0.15);
                margin: 0 -21px;
                padding: 0 21px;
                a {
                  color: #666;
                }
                & + li {
                  border-top-color: white;
                }
              }
            }
            &::before {
              @include dropdown-menu-top-arrow(#f3f3f3, 17px, calc(100% + 1px), 35px);
            }
            &::after {
              @include dropdown-menu-top-arrow(#fff, 15px, 100%, 37px);
            }
          }
        }
      }
      @media (max-width: 767px) {
        border-width: 0;
        border-bottom-width: 1px;
        border-radius: 0;
        .navbar-toggle {
          &.collapsed {
            background: transparent !important;
          }
          &:not(.collapsed), &:hover {
            background: #ddd;
          }
        }
        li {
          @include menu-border-top;
          a {
            .caret {
              display: none;
              // border-right: 7px solid transparent;
              // border-top: 7px dashed;
              // border-left: 7px solid transparent;
              // position: absolute;
              // right: 15px;
              // top: 21px;
            }
          }
          
        }
        a, .sign-out-btn {
          padding: 15px;
        }
        .navbar-nav {
          margin: 50px -15px 0;
        }
        .account-menu {
          .dropdown-toggle {
            border-bottom: 1px solid #e7e7e7;
          }
          .dropdown-menu {
            background-color: transparent;
            border: 0;
            border-radius: 0;
            box-shadow: none;
            display: block;
            float: none;
            padding: 0;
            position: static;
            margin-top: 0;
            width: auto;
            li {
              @include menu-border-top;
              a {
                line-height: 20px;
                padding-left: 40px;
                .notification-badge {
                  right: 11px;
                  top: 15px;
                }
              }
              .sign-out-btn {
                padding-left: 40px;
              }
            }
          }
        }
      }
    }
  }
}
