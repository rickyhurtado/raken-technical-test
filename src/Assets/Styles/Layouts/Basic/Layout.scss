.layout {
  > main.container {
    min-height: 300px;
    padding: 0 15px;
    width: 100%;
  }
}
