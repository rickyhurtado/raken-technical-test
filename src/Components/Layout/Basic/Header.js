import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { Navbar, NavDropdown } from 'react-bootstrap';
import { NavLink } from '../../../Lib/Common/Views';
import Button from '../../Button';

class Header extends Component {
  render() {
    const path = this.props.match.path;

    return (
      <header className="header">
        <Navbar>
          <Navbar.Header>
            <Link to="/" className="navbar-brand">{process.env.REACT_APP_SITE_NAME}</Link>
            <Navbar.Toggle id="js-navbar-toggle-btn" />
          </Navbar.Header>
          <Navbar.Collapse>
            <ul className="navbar-nav nav navbar-right">
              <NavLink title="Dashboard" to="/dashboard" path={path} />
              <NavLink title="Projects" to="/projects" path={path} />
              <NavLink title="Team" to="/team" path={path} />
              <NavLink title="Company" to="/company" path={path} />
              <NavDropdown className="account-menu" title="Account" id="js-account-menu">
                <NavLink title="My Profile" to="/account/my-profile" path={path} />
                <NavLink title="My Projects" to="/account/my-projects" path={path} count={1} />
                <NavLink title="Notifications" to="/account/notifications" path={path} count={10} />
                <NavLink title="Tasks" to="/account/tasks" path={path} count={100} />
                <NavLink title="Help" to="/help" path={path} />
                <Button.SignOut />
              </NavDropdown>
            </ul>
          </Navbar.Collapse>
        </Navbar>
      </header>
    );
  }
};

export default withRouter(Header);
