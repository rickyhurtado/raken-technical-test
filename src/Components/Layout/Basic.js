import React, { Component } from 'react';
import Header from './Basic/Header';

export default class Basic extends Component {
  render() {
    return (
      <div className="layout">
        <Header />
        <main className="container">
          {this.props.children}
        </main>
      </div>
    );
  }
};
