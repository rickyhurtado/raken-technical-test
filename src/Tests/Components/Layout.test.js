import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Layout from '../../Components/Layout';
import Header from  '../../Components/Layout/Basic/Header';

it('Layout.Basic component should render', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Route>
        <Layout.Basic>
          <p>Test</p>
        </Layout.Basic>
      </Route>
    </BrowserRouter>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('Header component should render', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Route>
        <Header />
      </Route>
    </BrowserRouter>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
