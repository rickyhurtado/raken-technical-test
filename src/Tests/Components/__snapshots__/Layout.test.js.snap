// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
<header
  className="header"
>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          Company Name
        </a>
        <button
          className="navbar-toggle collapsed"
          id="js-navbar-toggle-btn"
          onClick={[Function]}
          type="button"
        >
          <span
            className="sr-only"
          >
            Toggle navigation
          </span>
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
          <span
            className="icon-bar"
          />
        </button>
      </div>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
      >
        <ul
          className="navbar-nav nav navbar-right"
        >
          <li
            className=""
          >
            <a
              href="/dashboard"
              onClick={[Function]}
            >
              Dashboard
               
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="/projects"
              onClick={[Function]}
            >
              Projects
               
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="/team"
              onClick={[Function]}
            >
              Team
               
            </a>
          </li>
          <li
            className=""
          >
            <a
              href="/company"
              onClick={[Function]}
            >
              Company
               
            </a>
          </li>
          <li
            className="account-menu dropdown"
            style={undefined}
          >
            <a
              aria-expanded={false}
              aria-haspopup={true}
              className="dropdown-toggle"
              href="#"
              id="js-account-menu"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
            >
              Account
               
              <span
                className="caret"
              />
            </a>
            <ul
              aria-labelledby="js-account-menu"
              className="dropdown-menu"
              role="menu"
            >
              <li
                className=""
              >
                <a
                  href="/account/my-profile"
                  onClick={[Function]}
                >
                  My Profile
                   
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/account/my-projects"
                  onClick={[Function]}
                >
                  My Projects
                   
                  <span
                    className="notification-badge"
                  >
                    1
                  </span>
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/account/notifications"
                  onClick={[Function]}
                >
                  Notifications
                   
                  <span
                    className="notification-badge"
                  >
                    10
                  </span>
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/account/tasks"
                  onClick={[Function]}
                >
                  Tasks
                   
                  <span
                    className="notification-badge"
                  >
                    99+
                  </span>
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/help"
                  onClick={[Function]}
                >
                  Help
                   
                </a>
              </li>
              <li>
                <button
                  className="sign-out-btn"
                >
                  Sign Out
                </button>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Layout.Basic component should render 1`] = `
<div
  className="layout"
>
  <header
    className="header"
  >
    <nav
      className="navbar navbar-default"
    >
      <div
        className="container"
      >
        <div
          className="navbar-header"
        >
          <a
            className="navbar-brand"
            href="/"
            onClick={[Function]}
          >
            Company Name
          </a>
          <button
            className="navbar-toggle collapsed"
            id="js-navbar-toggle-btn"
            onClick={[Function]}
            type="button"
          >
            <span
              className="sr-only"
            >
              Toggle navigation
            </span>
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
          </button>
        </div>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
        >
          <ul
            className="navbar-nav nav navbar-right"
          >
            <li
              className=""
            >
              <a
                href="/dashboard"
                onClick={[Function]}
              >
                Dashboard
                 
              </a>
            </li>
            <li
              className=""
            >
              <a
                href="/projects"
                onClick={[Function]}
              >
                Projects
                 
              </a>
            </li>
            <li
              className=""
            >
              <a
                href="/team"
                onClick={[Function]}
              >
                Team
                 
              </a>
            </li>
            <li
              className=""
            >
              <a
                href="/company"
                onClick={[Function]}
              >
                Company
                 
              </a>
            </li>
            <li
              className="account-menu dropdown"
              style={undefined}
            >
              <a
                aria-expanded={false}
                aria-haspopup={true}
                className="dropdown-toggle"
                href="#"
                id="js-account-menu"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                Account
                 
                <span
                  className="caret"
                />
              </a>
              <ul
                aria-labelledby="js-account-menu"
                className="dropdown-menu"
                role="menu"
              >
                <li
                  className=""
                >
                  <a
                    href="/account/my-profile"
                    onClick={[Function]}
                  >
                    My Profile
                     
                  </a>
                </li>
                <li
                  className=""
                >
                  <a
                    href="/account/my-projects"
                    onClick={[Function]}
                  >
                    My Projects
                     
                    <span
                      className="notification-badge"
                    >
                      1
                    </span>
                  </a>
                </li>
                <li
                  className=""
                >
                  <a
                    href="/account/notifications"
                    onClick={[Function]}
                  >
                    Notifications
                     
                    <span
                      className="notification-badge"
                    >
                      10
                    </span>
                  </a>
                </li>
                <li
                  className=""
                >
                  <a
                    href="/account/tasks"
                    onClick={[Function]}
                  >
                    Tasks
                     
                    <span
                      className="notification-badge"
                    >
                      99+
                    </span>
                  </a>
                </li>
                <li
                  className=""
                >
                  <a
                    href="/help"
                    onClick={[Function]}
                  >
                    Help
                     
                  </a>
                </li>
                <li>
                  <button
                    className="sign-out-btn"
                  >
                    Sign Out
                  </button>
                </li>
              </ul>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </header>
  <main
    className="container"
  >
    <p>
      Test
    </p>
  </main>
</div>
`;
